validator/src/main.rs

```diff
@@ -1382,11 +1382,9 @@ pub fn main() {
                 usize
             ),
             worker_threads: value_t_or_exit!(matches, "rpc_pubsub_worker_threads", usize),
-            notification_threads: if full_api {
-                value_of(&matches, "rpc_pubsub_notification_threads")
-            } else {
-                Some(0)
-            },
+            notification_threads: value_t!(matches, "rpc_pubsub_notification_threads", usize)
+                .ok()
+                .and_then(NonZeroUsize::new),
```

Comment:
Use of `.requires()` and `.default_value_if()` allow us to simplify this check. The scenarios are
```
full-rpc | n-thread | n-thread value
------------------------------------
  false  |   false  | value_t() will yield a None
  false  |   true   | CLAP errors out
  true   |   false  | value_t() will match the default
  true   |   true   | value_t() will match N (user value)
```
where:
- `full-rpc` ==> `--full-rpc-api` passed
- `n-thread` ==> `--rpc-pubsub-notification-threads N` passed
- `n-thread value` ==> The resulting value that is parsed and set in `pubsub_config.notification_threads`